core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1180,6 +1180,14 @@ package object config {
       .intConf
       .createWithDefault(1)
 
+  private[spark] val IO_WARNING_LARGEFILETHRESHOLD =
+    ConfigBuilder("spark.io.warning.largeFileThreshold")
+      .internal()
+      .doc("When spark loading one single large file, if file size exceed this " +
+        "threshold, then log warning with possible reasons.")
+      .longConf
+      .createWithDefault(1024 * 1024 * 1024)
```

Comment:
I don't think it's worth adding a config. It looks an overkill.

Reply:
1. this is an internal config. 2. "large file" is vague, and I don't think we can hardcode a value and say that's "large file".

Reply:
The problem is, this warning stuff is trivial and not important actually.

We can just pick any reasonable number. Who will configure this? I won't do that. This information shouldn't be job-based, too.

Reply:
people may set it to Long.Max to disable the warning. Besides, an internal config doesn't hurt. I think we have many internal configs that users will never set.